testing/asserts.ts

```diff
@@ -864,3 +866,59 @@ export function unimplemented(msg?: string): never {
 export function unreachable(): never {
   throw new AssertionError("unreachable");
 }
+
+let snapshotFile: Record<string, unknown> | undefined = undefined;
+const updatedSnapshotFile: Record<string, unknown> = {};
+// const snapshotMap: Record<string, number> = {};
+
+export async function assertSnapshot(context: Deno.TestContext, actual: unknown) {
+  const name = getName(context);
+  // const count = getCount(name);
+
+  if (!snapshotFile) {
+    const snapshotPath = getSnapshotPath();
+    await ensureFile(snapshotPath);
+    const file = await Deno.readTextFile(snapshotPath);
+    snapshotFile = file ? JSON.parse(file) : {};
+  }
+  const snapshot = snapshotFile?.[name];
+  if (!Deno.args.includes('--update')) {
+    assertEquals(actual, snapshot);
+  } else {
+    let isEqual = true;
+    try {
+      assertEquals(actual, snapshot);
+    } catch {
+      isEqual = false;
+    }
+    if (!isEqual) console.info("Snapshot updated", name);
+  }
+  updatedSnapshotFile[name] = actual;
```

Comment:
Does this mean this implementation supports only 1 snapshot per test case?

Reply:
Does the test case mean one `Deno.test`?
This implementation supports 1 snapshot per test module (one .test.ts file)

for example, if I test `a.test.ts` with `--update` arg,
```typescript
Deno.test("Snapshot Test A", async (t) => {
  await assertSnapshot(t, { a: 1, b: 2 });
  await t.step("yaho", async (t) => {
    await assertSnapshot(t, { b: 2, c: 3 });
    await t.step("haha", async (t) => {
      await assertSnapshot(t, { b: 2, c: 4 });
    });
  });
});

Deno.test("Snapshot Test A2", async (t) => {
  await assertSnapshot(t, { a: 1, b: 2 });
  await t.step("what", async (t) => {
    await assertSnapshot(t, { b: 2, c: 3 });
    await t.step("good", async (t) => {
      await assertSnapshot(t, { b: 2, c: 4 });
    });
  });
});
```

the output, `a.test.snap` will be like this
```json
{
  "Snapshot Test A": {
    "a": 1,
    "b": 2
  },
  "Snapshot Test A > yaho": {
    "b": 2,
    "c": 3
  },
  "Snapshot Test A > yaho > haha": {
    "b": 2,
    "c": 4
  },
  "Snapshot Test A2": {
    "a": 1,
    "b": 2
  },
  "Snapshot Test A2 > what": {
    "b": 2,
    "c": 3
  },
  "Snapshot Test A2 > what > good": {
    "b": 2,
    "c": 4
  }
}
```

Reply:
How about the below case? I think the second `assertSnapshot` call overwrites the snapshot from the first call

```js
Deno.test("Snapshot Test A", async (t) => {
  await assertSnapshot(t, { a: 1, b: 2 });
  await assertSnapshot(t, { c: 3, d: 4 });
});
```

Reply:
Aha! Sorry, now I know what you mean. That case will be handled with commented count. (Currently not implemented perfectly so I commented) @kt3k

Reply:
Ah, ok. I see. `getCount` does that 👍